File: flutter_app/lib/solver/division.dart
import 'operation.dart';

class Division implements Operation {
  const Division();

  @override
  double calc(double a, double b) => b == 0.0 ? double.nan : a / b;

  @override
  String format(String a, String b) => '$a/$b';

  // Matching Python original: set to true.
  // Set to false to find all solutions including reversed operand order.
  @override
  bool get isCommutative => false;

  @override
  int get priority => 1;

  @override
  String get symbol => '/';
}

File: Operations/division.py
from Operations.operation import Operation


class Division(Operation):

    @staticmethod
    def calc(value1, value2):
        if value2 == 0:
            return float('nan')
        return value1 / value2

    @classmethod
    @property
    def is_commutative(cls):
        return True

    @classmethod
    @property
    def priority(cls):
        return 0

    @staticmethod
    def to_string(value1, value2):
        return value1 + "/" + value2